pdc_client/__init__.py

```diff
@@ -186,35 +157,21 @@ class to perform requests.
         if not server:
             raise TypeError('Server must be specified')
         self.session = requests.Session()
-        config = read_config_file(server)
 
-        # Command line must *always* override configuration
-        if config:
-            try:
-                url = config[CONFIG_URL_KEY_NAME]
-            except KeyError:
-                sys.stderr.write("'{}' must be specified in configuration for '{}'".format(
-                    CONFIG_URL_KEY_NAME, server))
-                sys.exit(1)
+        try:
+            config = server_configuration(server)
+            url = config.url()
 
+            # Command line must *always* override configuration
             if ssl_verify is None:
-                cfg_ssl_verify = config.get(CONFIG_SSL_VERIFY_KEY_NAME)
-                insecure = config.get(CONFIG_INSECURE_KEY_NAME)
-                if insecure is not None:
-                    sys.stderr.write("Warning: '{}' option is deprecated; please use '{}' "
-                                     "instead\n".format(
-                                         CONFIG_INSECURE_KEY_NAME, CONFIG_SSL_VERIFY_KEY_NAME))
-                    ssl_verify = not insecure
-                if cfg_ssl_verify is not None:
-                    ssl_verify = cfg_ssl_verify
+                ssl_verify = config.ssl_verify()
             if develop is None:
-                develop = config.get(CONFIG_DEVELOP_KEY_NAME, develop)
+                develop = config.is_development()
             if token is None:
-                token = config.get(CONFIG_TOKEN_KEY_NAME, token)
-        else:
-            url = server
-            if ssl_verify is None:
-                ssl_verify = True
+                token = config.token()
+        except ServerConfigError as e:
+            logger.error(e)
+            sys.exit(1)
```

Comment:
I think this would be better handled in the module that imports `PDCClient`. If someone is using it from their python script, they would not be able to react to this error: it would just kill their script.

Reply:
That would break the API in some strange way. But ... I guess calling exit here is a bug so I'm fixing it.